src/core/hle/service/caps/caps_a.cpp

```diff
@@ -62,138 +62,70 @@ IAlbumAccessorService::IAlbumAccessorService(Core::System& system_,
 
 IAlbumAccessorService::~IAlbumAccessorService() = default;
 
-void IAlbumAccessorService::DeleteAlbumFile(HLERequestContext& ctx) {
-    IPC::RequestParser rp{ctx};
-    const auto file_id{rp.PopRaw<AlbumFileId>()};
-
+Result IAlbumAccessorService::DeleteAlbumFile(AlbumFileId file_id) {
     LOG_INFO(Service_Capture, "called, application_id=0x{:0x}, storage={}, type={}",
              file_id.application_id, file_id.storage, file_id.type);
 
-    Result result = manager->DeleteAlbumFile(file_id);
-    result = TranslateResult(result);
-
-    IPC::ResponseBuilder rb{ctx, 2};
-    rb.Push(result);
+    const Result result = manager->DeleteAlbumFile(file_id);
+    R_RETURN(TranslateResult(result));
 }
 
-void IAlbumAccessorService::IsAlbumMounted(HLERequestContext& ctx) {
-    IPC::RequestParser rp{ctx};
-    const auto storage{rp.PopEnum<AlbumStorage>()};
-
+Result IAlbumAccessorService::IsAlbumMounted(Out<bool> out_is_mounted, AlbumStorage storage) {
     LOG_INFO(Service_Capture, "called, storage={}", storage);
 
-    Result result = manager->IsAlbumMounted(storage);
-    const bool is_mounted = result.IsSuccess();
-    result = TranslateResult(result);
-
-    IPC::ResponseBuilder rb{ctx, 3};
-    rb.Push(result);
-    rb.Push<u8>(is_mounted);
+    const Result result = manager->IsAlbumMounted(storage);
+    *out_is_mounted = result.IsSuccess();
+    R_RETURN(TranslateResult(result));
 }
 
-void IAlbumAccessorService::Unknown18(HLERequestContext& ctx) {
-    struct UnknownBuffer {
-        INSERT_PADDING_BYTES(0x10);
-    };
-    static_assert(sizeof(UnknownBuffer) == 0x10, "UnknownBuffer is an invalid size");
-
+Result IAlbumAccessorService::Unknown18(
+    Out<u32> out_buffer_size,
+    OutArray<u8, BufferAttr_HipcMapAlias | BufferAttr_HipcMapTransferAllowsNonSecure> out_buffer) {
     LOG_WARNING(Service_Capture, "(STUBBED) called");
-
-    std::vector<UnknownBuffer> buffer{};
-
-    if (!buffer.empty()) {
-        ctx.WriteBuffer(buffer);
-    }
-
-    IPC::ResponseBuilder rb{ctx, 3};
-    rb.Push(ResultSuccess);
-    rb.Push(static_cast<u32>(buffer.size()));
+    out_buffer_size = 0;
```

Comment:
```suggestion
    *out_buffer_size = 0;
```
If this actually compiles I should fix it so that it doesn't